Gems/ROS2/Code/Source/SdfAssetBuilder/SdfAssetBuilder.cpp

```diff
@@ -184,6 +147,25 @@ namespace ROS2
         return assetMap;
     }
 
+    AZStd::string SdfAssetBuilder::GetFingerprint() const
+    {
+        AZStd::string settingsString;
+        AZ::IO::ByteContainerStream<AZStd::string> stream { &settingsString };
+
+        // Always write all of the settings, even when they match the defaults.
+        // This isn't strictly necessary, but this way if the default values ever get changed
+        // we'll still be able to detect that the settings don't match what they used to be.
+        AZ::JsonSerializerSettings jsonSettings;
+        jsonSettings.m_keepDefaults = true;
+
+        [[maybe_unused]] AZ::Outcome<void, AZStd::string> saveObjectResult =
+            AZ::JsonSerializationUtils::SaveObjectToStream(&m_globalSettings, stream, {}, &jsonSettings);
+        AZ_Assert(saveObjectResult.IsSuccess(), "Failed to save settings to fingerprint string: %s", 
+            saveObjectResult.GetError().c_str());
+
+        return settingsString;
```

Comment:
The fingerprint is the entire settings string? not a hash?

Reply:
I am fine with it the way it is, its just that reading `fingerprint` implies a hash.